### PR TITLE
Feature/m2 1489 add validation for subscale section

### PR DESCRIPTION
**Pull request by alex-i-e**

### CODE REVIEW

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/TitleComponent/TitleComponent.styles.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { styled, Box } from '@mui/material';
+import { theme, variables } from 'shared/styles';
+
+export const StyledMark = styled(Box)`
+  position: absolute;
+  left: -2rem;
+  border-radius: 100%;
```
Comment (muhurov) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/TitleComponent/TitleComponent.styles.ts:7]: please use variable for border-radius

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.tsx [hunk 2]
```diff
@@ -63,32 +66,46 @@ export const SubscalesConfiguration = () => {
 
   const handleOnTotalScoreLookupTable = () => false;
 
+  useEffect(() => {
+    if (calculateTotalScoreSwitch) {
+      register(calculateTotalScoreName);
+      setValue(calculateTotalScoreName, SubscaleTotalScore.Sum);
+    }
+    if (!calculateTotalScoreSwitch) {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/SubscalesConfiguration.tsx:74]: maybe it's better to use else here?

File: src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/TitleComponent/TitleComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  variables,
+} from 'shared/styles';
+
+import { TitleComponentProps } from './TitleComponent.types';
+import { StyledMark } from './TitleComponent.styles';
+
+export const TitleComponent = ({ title, name, open }: TitleComponentProps) => {
+  const { getFieldState } = useFormContext();
+  const errorObject = getFieldState(name).error as unknown as Record<string, FieldError>;
+  const hasErrors = !!getFieldState(name).error;
```
Comment (IhorNesterenko99) [inline src/modules/Builder/features/ActivitySettings/SubscalesConfiguration/TitleComponent/TitleComponent.tsx:17]: Why not to use !!errorObject ?

File: src/shared/components/SwitchWithState/SwitchWithState.types.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
```
Comment (IhorNesterenko99) [inline src/shared/components/SwitchWithState/SwitchWithState.types.ts:1]: Could you pls import ChangeEvent from react?

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by muhurov — APPROVED** [review]

**Pull request merged**